### PR TITLE
SNOW-1555315: Support column lineage in Session.lineage.trace API.

### PR DESCRIPTION
**Pull request by sfc-gh-rsureshbabu**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

Adds support for column lineage in the DataFrame.lineage.trace API.
   

   Fixes SNOW-1555315 

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

File: tests/integ/test_lineage.py [hunk 1]
```diff
@@ -247,6 +246,37 @@ def test_lineage_trace(session):
         "DISTANCE": [1, 2],
     }
 
+    expected_df = pd.DataFrame(expected_data)
+    # TODO Enable the check after SNOW-1437475 is fixed.
+    # assert_frame_equal(df, expected_df, check_dtype=False)
+
+    # CASE 7 : Column lineage
+    session.sql(
+        f'CREATE OR REPLACE VIEW {db}.{schema}."v7" AS SELECT * FROM {db}.{schema}.V2'
```
Comment (sfc-gh-tbao) [inline tests/integ/test_lineage.py:255]: shall we cover a case where column project happens here?

Comment (sfc-gh-tbao) [inline tests/integ/test_lineage.py:255]: where are we checking the lineage flow for v7?

File: tests/integ/test_lineage.py [hunk 2]
```diff
@@ -247,6 +246,37 @@ def test_lineage_trace(session):
...
+    ).collect()
+
+    df = session.lineage.trace(
+        f"{db}.{schema}.V2.C1", "COLUMN", direction=LineageDirection.UPSTREAM
```
Comment (sfc-gh-tbao) [inline tests/integ/test_lineage.py:259]: is it possible to bulk trace all the columns for a given obj? e.g. 
```
session.lineage.trace(
        f"{db}.{schema}.V2", "COLUMN", direction=LineageDirection.UPSTREAM
```
which returns all column lineages

Comment (sfc-gh-rsureshbabu) [reply tests/integ/test_lineage.py:259]: Discussed offline. Thats not supported right now.

File: tests/integ/test_lineage.py (hunk 1 above)
Comment (sfc-gh-rsureshbabu) [reply tests/integ/test_lineage.py:255]: Do you mean "v7".C1 ?

Comment (sfc-gh-rsureshbabu) [reply tests/integ/test_lineage.py:255]: v7 is not needed. Removed it

Comment (sfc-gh-tbao) [reply tests/integ/test_lineage.py]: select cols... from

File: src/snowflake/snowpark/lineage.py [hunk 3]
```diff
@@ -527,16 +539,18 @@ def _get_result_dataframe(
...
-            Case-sensitive: "\"database\".\"schema\".\"object\""
+            Non-Case-sensitive: "database.schema.object" or "database.schema.object.column_name"
+            Case-sensitive: "\"database\".\"schema\".\"object\"" or "\"database\".\"schema\".\"object\"".\"column_name\"
         """
         parts = _DGQLQueryBuilder.split_fully_qualified_name(object_name)
 
-        # Check if the object name has three parts separated by dots
-        if len(parts) != 3:
+        is_column_domain = object_domain.upper() == _SnowflakeDomain.COLUMN
+        if (is_column_domain and len(parts) != 4) or (
```
Comment (sfc-gh-yli) [inline src/snowflake/snowpark/lineage.py:551]: I didn't know that this API checks that it has to be fully qualified names, can't the caller just pass in partial names like 't1.c1' while the rest can be inferred from the current context? at least DGQL can support that

File: tests/integ/test_lineage.py [hunk 4]
```diff
@@ -247,6 +246,33 @@ def test_lineage_trace(session):
...
+    )
+
+    # Removing 'creadtedOn' field since the value can not be predicted.
+    df = remove_created_on_field(df.to_pandas())
+
+    expected_data = {
+        "SOURCE_OBJECT": [
+            {"domain": "COLUMN", "name": f"{db}.{schema}.V1.C1", "status": "ACTIVE"},
+            {"domain": "COLUMN", "name": f"{db}.{schema}.T1.C1", "status": "ACTIVE"},
```
Comment (sfc-gh-yli) [inline tests/integ/test_lineage.py:264]: do we need to include a new filed in the output for column's parent table domain? so we can indicate if it is view, featureView, etc. DGQL result already has the `tableType` field in `properties`

File: src/snowflake/snowpark/lineage.py (hunk 3 above)
Comment (sfc-gh-rsureshbabu) [reply src/snowflake/snowpark/lineage.py:551]: We can make the optimization as a follow up.

Comment (sfc-gh-yli) [reply src/snowflake/snowpark/lineage.py:551]: sg, not blocking on this

File: tests/integ/test_lineage.py (hunk 1 above)
Comment (sfc-gh-rsureshbabu) [reply tests/integ/test_lineage.py]: changed the definition of the view to use projection.

File: tests/integ/test_lineage.py (hunk 4 above)
Comment (sfc-gh-rsureshbabu) [reply tests/integ/test_lineage.py]: Good point. Exposing it as filed name "type" to consistent with srcType and dstType in SQL API

**Review by sfc-gh-yli — APPROVED** [review]
LGTM

**Review by sfc-gh-jrose — APPROVED** [review]

**Pull request merged**